devbox.go

```diff
@@ -175,7 +180,11 @@ func (d *Devbox) Shell() error {
 		return errors.WithStack(err)
 	}
 	nixShellFilePath := filepath.Join(d.srcDir, ".devbox/gen/shell.nix")
-	sh, err := nix.DetectShell(nix.WithPlanInitHook(plan.ShellInitHook), nix.WithProfile(d.profileDir()))
+	sh, err := nix.DetectShell(
+		nix.WithPlanInitHook(plan.ShellInitHook),
+		nix.WithProfile(d.profileDir()),
+		nix.WithHistoryFile(filepath.Join(d.srcDir, shellHistoryFile)),
```

Comment:
The current design maintains a strict separation between the outside shell history and the devbox shell history.

Should we be respecting any existing shell history file?

That would involve a small change like so:

```
nix.WithHistoryFile(d.shellHistoryFile()),
```

and
```
func (d *Devbox) shellHistoryFile() string {
   envHistFile := os.Getenv("HISTFILE")
   if envHistFile != "" {
       return envHistFile
   }
   return filepath.Join(d.srcDir, shellHistoryFile)
}
```

WDYT? 
I'm inclined to keep the current implementation (i.e. maintaining the strict separation of histories), but wanted to have the discussion.

Reply:
I think keeping them separate makes sense, but this is also probably one of those UX things where we won't know for sure until we use it for a while.

Reply:
My intuition is that each devbox project has its own history is 👍 👍 👍

Reply:
I found this convo while considering logging a feature request to use (and append to) existing shell history. Before devbox I made a decision to go with asdf over dev containers or nix at my company for the primary reason that us devs are persnickety about our shells. I'd like devbox to fade into the background as much as possible and this feels like a rough edge to me.

Reply:
@mjgallag thanks for raising this topic. I filed an issue to track your request and have a discussion: https://github.com/jetpack-io/devbox/issues/939